### PR TITLE
Zigbee2mqtt - Add light color feature

### PR DESCRIPTION
**Pull request by cicoub13**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [x] If your changes modify the API (REST or Node.js), did you modify the API documentation? (Documentation is based on comments in code)
- [x] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [x] Did you add fake requests data for the demo mode (`front/src/config/demo.json`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Adding Zigbee color XY management for compatible devices (get color and set color)

### CODE REVIEW

File: server/utils/colors.js [hunk 1]
```diff
@@ -64,9 +64,68 @@ function hexToInt(hexColor) {
...
+function xyToInt(x, y) {
+  /**
+   * @description Reverse Gamma correction applied in rgbToXy.
+   * @param {number} value - Color value.
+   * @returns {number} ReversedGammaCorrectedValue - Color with Gamma added.
+   * @example
+   * const value = getReversedGammaCorrectedValue(0,5);
+   * console.log(value === 0.73535698305);
+   */
+  function getReversedGammaCorrectedValue(value) {
```
Comment (atrovato) [inline server/utils/colors.js]: Maybe this method can out of `xyToInt`, as a const one?

**Review by atrovato — COMMENTED** [review]
Nice, just one remark to move a method :)

File: server/utils/colors.js (hunk 1 above)
Comment (cicoub13) [reply server/utils/colors.js]: Like it's very dedicated to the `xyToInt` function, I preferred to keep it inside (not usable outside the function).
Do you think it's better to put it outside ?

Comment (atrovato) [reply server/utils/colors.js]: It's to avoid to declare same function each time we enter in upper method.

Comment (Pierre-Gilles) [reply server/utils/colors.js]: +1, as this function doesn't use variable from xyToInt, it doesn't need to be re-created at each call to xyToInt. It's better to create it outside!

**Comment by Pierre-Gilles** [comment]
Very nice PR ! 🙂 Let's wait for feedbacks on this PR before merging this, otherwise it looks good to me!

File: server/utils/colors.js (hunk 1 above)
Comment (cicoub13) [reply server/utils/colors.js]: Fixed

**Comment by cicoub13** [comment]
I added the Light Temperature Box in this commit https://github.com/GladysAssistant/Gladys/pull/1203/commits/afb53ce926411b53b6cc4f9604e07cd239e87f33

Can you have a look to the code ?

**Comment by Pierre-Gilles** [comment]
@cicoub13 Looks good to me! Tested it with the MQTT integration and it works perfectly :) Is this PR ready to merge then? :)

**Comment by Pierre-Gilles** [comment]
Oh, and while we are talking about the slider, any idea why the existing slide doesn't work well on mobile ? ( https://github.com/GladysAssistant/Gladys/issues/1167 )

I just tested with the latest version of tabler, and it seems they fixed the slider: https://preview.tabler.io/docs/form-elements.html

Any idea what's wrong in our current CSS?

**Comment by cicoub13** [comment]
> @cicoub13 Looks good to me! Tested it with the MQTT integration and it works perfectly :) Is this PR ready to merge then? :)

Tested and ready to be merged 🚀

**Comment by cicoub13** [comment]
> Oh, and while we are talking about the slider, any idea why the existing slide doesn't work well on mobile ? ( #1167 )
> 
> I just tested with the latest version of tabler, and it seems they fixed the slider: https://preview.tabler.io/docs/form-elements.html
> 
> Any idea what's wrong in our current CSS?

I will have a look

**Comment by Pierre-Gilles** [comment]
> Tested and ready to be merged 🚀

Amazing! I'll merge this now then 🚀

> I will have a look

Ok, let me know if you find something, we can fix this in a separate PR :) 

Thanks again for all the work here!

**Review by Pierre-Gilles — APPROVED** [review]

**Pull request merged**